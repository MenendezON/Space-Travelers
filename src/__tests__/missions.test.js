import React from 'react';
import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import Missions from '../components/missions';

// Mocking react-redux's useSelector
jest.mock('react-redux');

// Mocking axios
jest.mock('axios', () => ({
  get: jest.fn(),
}));

test('Missions component renders correctly', () => {
  // Mock the useSelector behavior
  useSelector.mockReturnValue([
    {
      mission_id: '9D1B7E0',
      mission_name: 'Thaicom',
      description: 'Thaicom is the name of a series of communications ... Thailand and modern communications technology.',
      reserved: true,
    },
    {
      mission_id: 'F4F83DE',
      mission_name: 'Telstar',
      description: 'Telstar 19V (Telstar 19 Vantage) is a communicatio ... launched by Ariane 5ECA on 1 July 2009.',
      reserved: true,
    },
    // Add more mission objects as needed
  ]);

  const { container } = render(<Missions />);

  expect(container).toMatchSnapshot();
});
