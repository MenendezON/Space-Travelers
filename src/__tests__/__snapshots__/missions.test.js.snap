// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions component renders correctly 1`] = `
<div>
  <div
    class="missions-page"
  >
    <h1>
      Missions
    </h1>
    <div
      class="mission-container"
    >
      <ul
        class="mission-set mission-head"
      >
        <li
          class="mission-name"
        >
          Mission Name
        </li>
        <li
          class="mission-description"
        >
          Description
        </li>
        <li
          class="mission-status"
        >
          Status
        </li>
        <li
          class="mission-action"
        >
          Action
        </li>
      </ul>
      <ul
        class="mission-set"
      >
        <li
          class="mission-name"
        >
          Thaicom
        </li>
        <li
          class="mission-description"
        >
          Thaicom is the name of a series of communications ... Thailand and modern communications technology.
        </li>
        <li
          class="mission-status"
        >
          <button
            class="membership_on"
            id="membership"
            type="submit"
          >
            Active Member
          </button>
        </li>
        <li
          class="mission-action"
        >
          <button
            class="join_on"
            id="join"
            type="submit"
          >
            Leave Mission
          </button>
        </li>
      </ul>
      <ul
        class="mission-set"
      >
        <li
          class="mission-name"
        >
          Telstar
        </li>
        <li
          class="mission-description"
        >
          Telstar 19V (Telstar 19 Vantage) is a communicatio ... launched by Ariane 5ECA on 1 July 2009.
        </li>
        <li
          class="mission-status"
        >
          <button
            class="membership_on"
            id="membership"
            type="submit"
          >
            Active Member
          </button>
        </li>
        <li
          class="mission-action"
        >
          <button
            class="join_on"
            id="join"
            type="submit"
          >
            Leave Mission
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
